src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -13274,14 +13274,15 @@ public sealed partial class FieldOffsetAttribute : System.Attribute
         public FieldOffsetAttribute(int offset) { }
         public int Value { get { throw null; } }
     }
-    public partial struct GCHandle
+    public partial struct GCHandle : System.IDisposable
     {
         private int _dummyPrimitive;
         public bool IsAllocated { get { throw null; } }
         public object? Target { get { throw null; } set { } }
         public System.IntPtr AddrOfPinnedObject() { throw null; }
         public static System.Runtime.InteropServices.GCHandle Alloc(object? value) { throw null; }
         public static System.Runtime.InteropServices.GCHandle Alloc(object? value, System.Runtime.InteropServices.GCHandleType type) { throw null; }
+        public void Dispose() { }
```

Comment:
Are there places in dotnet/runtime we're currently manually using a try/finally with Free that we could now switch to a using?

Reply:
Second pattern to look for: `if (gcHandle.IsAllocated) gcHandle.Free()`.

Reply:
I've migrated the existing patterns I've found in https://github.com/dotnet/runtime/pull/55596/commits/97b0ddd5b87a44887a7431327d950e5f4a13fed7. For refactorings to `using GCHandle`, I've only updated patterns that exactly followed the `try/finally` bit that the `using` generates, to be extra safe (at least for now). I have a partially related question: while digging through the runtime to find code to update, I've noticed a bunch of usages of unguarded `GCHandle`-s in the `/Speech` subirectory, like this:

https://github.com/dotnet/runtime/blob/a38fc2fd272313165e30f8301d4d767a5c30f659/src/libraries/System.Speech/src/Internal/Synthesis/AudioBase.cs#L408-L415

Isn't this technically not safe and potentially introducing a leak?
Or is this fine in this specific scenario for some other reason? 🤔

Reply:
> Isn't this technically not safe and potentially introducing a leak?

Yes, seems like an OOM or similar failure would result in a leak.  We've basically not touched System.Speech other than getting it running on .NET Core. cc: @danmoseley

Reply:
> I've migrated the existing patterns I've found in 97b0ddd.

That doesn't look like the right commit?

Reply:
Sorry, misclicked the push button in the GitHub app 😅
The right commit is https://github.com/dotnet/runtime/pull/55596/commits/602b50c457083d226a147b3ecbeafd2ebbf14cf6.

Reply:
Please open a separate issue for the Speech code if necessary. That code has likely existed that way for 10+ years, so presumably any impact of the bug is pretty low.